google/cloud/storage/client.h

```diff
@@ -79,6 +79,39 @@ inline namespace STORAGE_CLIENT_NS {
  *     gcs::ClientOptions(gcs::oauth2::CreateAnonymousCredentials()));
  * @endcode
  *
+ * @par Error Handling
```

Comment:
I wonder if StatusOr documentation like this would be better left to the status_or.h file (and any online developer documentation). Otherwise, we may end up with lots of duplicate documentation in all the public .h files that return StatusOr. What do you think?

Reply:
I defer to you and @frankyn  on this one.  I expect we would have one of these for each major class, and (typically) we have just one major class per API (Bigtable uses "old style" per-resource objects, but there is only one `Client` for storage, and we will likely do the same for the other APIs).

In case it helps, I uploaded the generated documentation:

https://coryan.github.io/google-cloud-cpp/latest/storage/classgoogle_1_1cloud_1_1storage_1_1v0_1_1Client.html#details

Maybe remove the examples and leave the "Error Handling" paragraph?

Reply:
Coming over from #1799. How about moving examples to the index.html page. I think back linking to the index.html in client.h for more information would help reduce redundancy and keep everything in one place.

There currently isn't a single source cloud.google.com/storage doc for handling exceptions/error across client libraries so the index.html would be it for now.